src/Drupal/DrupalKernel.php

```diff
@@ -124,4 +128,36 @@ protected function addDrushServiceProvider($serviceProviderName, $filename)
             $this->serviceYamls['app'][$serviceProviderName] = $filename;
         }
     }
+
+    /**
+     * populates theme data on the filesystem.
+     *
+     * @see Drupal\Core\DrupalKernel::moduleData().
+     */
+    protected function themeData()
+    {
+        // First, find profiles.
+        $listing = new ExtensionDiscovery($this->root);
+        $listing->setProfileDirectories([]);
+        $all_profiles = $listing->scan('profile');
+        $profiles = array_intersect_key($all_profiles, $this->moduleList);
+
+        // If a module is within a profile directory but specifies another
+        // profile for testing, it needs to be found in the parent profile.
+        $settings = $this->getConfigStorage()->read('simpletest.settings');
+        $parent_profile = !empty($settings['parent_profile']) ? $settings['parent_profile'] : NULL;
+        if ($parent_profile && !isset($profiles[$parent_profile])) {
+            // In case both profile directories contain the same extension, the
+            // actual profile always has precedence.
+            $profiles = array($parent_profile => $all_profiles[$parent_profile]) + $profiles;
+        }
+
+        $profile_directories = array_map(function ($profile) {
+            return $profile->getPath();
+        }, $profiles);
+        $listing->setProfileDirectories($profile_directories);
+
+        // Now find themes.
+        $this->moduleData += $listing->scan('theme');
```

Comment:
Same comment - not sure that it is safe to modify `$this->moduleData` like this.